spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -228,20 +287,20 @@ non-determinism must be fixed and the nodes restarted.
 
 Sources of non-determinism in applications may include:
 
-* Hardware failures
-    * Cosmic rays, overheating, etc.
-* Node-dependent state
-    * Random numbers
-    * Time
-* Underspecification
-    * Library version changes
-    * Race conditions
-    * Floating point numbers
-    * JSON or protobuf serialization
-    * Iterating through hash-tables/maps/dictionaries
-* External Sources
-    * Filesystem
-    * Network calls (eg. some external REST API service)
+- Hardware failures
+    - Cosmic rays, overheating, etc.
```

Comment:
Lol

Reply:
:-)

Reply:
Be careful when running Tendermint on a spacecraft.

Reply:
"Issue 10499232331: Intergalactic gossip network stalls due to unreachable peers inside or behind spacetime singularities."

Reply:
That's the problem when you locate your full nodes on the dark side of the moon!